packages/theme/src/cli/services/check.test.ts

```diff
@@ -0,0 +1,307 @@
+    const offenses: Offense[] = [
+      {
+        type: SourceCodeType.LiquidHtml,
+        check: 'LiquidHTMLSyntaxError',
+        message: 'Attempting to close HtmlElement',
+        absolutePath: '/path/to/file',
+        severity: Severity.ERROR,
+        start: {index: 0, line: 1, character: 0},
+        end: {index: 10, line: 1, character: 10},
+      },
+      {
+        type: SourceCodeType.LiquidHtml,
+        check: 'LiquidHTMLSyntaxError',
+        message: 'Attempting to close HtmlElement',
+        absolutePath: '/path/to/file',
+        severity: Severity.WARNING,
+        start: {index: 0, line: 2, character: 0},
+        end: {index: 10, line: 2, character: 10},
+      },
+    ]
+    const theme: unknown = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+    const result = formatSummary(offenses, theme as Theme)
+
+    expect(result).toEqual(['10 files inspected', 'with 2 total offenses found.', '\n1 errors.', '\n1 suggestions.'])
+  })
+})
+
+describe('renderOffensesText', () => {
+  beforeEach(() => {
+    const readFileMock = fs.readFileSync as Mock
+    readFileMock.mockReturnValue('Line1\nLine2\nLine3')
+  })
+
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  afterAll(() => {
+    vi.unstubAllGlobals()
+  })
+
+  it('should call renderError for offenses with severity ERROR', () => {
+    const offensesByFile = {
+      '/path/to/file': [
+        {
+          type: SourceCodeType.LiquidHtml,
+          check: 'LiquidHTMLSyntaxError',
+          message: 'Attempting to close HtmlElement',
+          absolutePath: '/path/to/file',
+          severity: Severity.ERROR,
+          start: {index: 0, line: 1, character: 0},
+          end: {index: 10, line: 1, character: 10},
+        },
+      ],
+    }
+    const themeRootPath = '/path/to'
+
+    renderOffensesText(offensesByFile, themeRootPath)
+
+    expect(renderError).toHaveBeenCalledTimes(1)
+    expect(renderWarning).toHaveBeenCalledTimes(0)
+  })
+
+  it('should call renderWarning for offenses with severity WARNING or INFO', () => {
+    const offensesByFile = {
+      '/path/to/file': [
+        {
+          type: SourceCodeType.LiquidHtml,
+          check: 'LiquidHTMLSyntaxError',
+          message: 'Attempting to close HtmlElement',
+          absolutePath: '/path/to/file',
+          severity: Severity.WARNING,
+          start: {index: 0, line: 1, character: 0},
+          end: {index: 10, line: 1, character: 10},
+        },
+      ],
+    }
+    const themeRootPath = '/path/to'
+
+    renderOffensesText(offensesByFile, themeRootPath)
+
+    expect(renderError).toHaveBeenCalledTimes(0)
+    expect(renderWarning).toHaveBeenCalledTimes(1)
+  })
+})
+describe('formatOffensesJson', () => {
+  it('should format offenses into JSON correctly', () => {
+    const offensesByFile = {
+      '/path/to/file': [
+        {
+          type: SourceCodeType.LiquidHtml,
+          check: 'LiquidHTMLSyntaxError',
+          message: 'Attempting to close HtmlElement',
```

Comment:
IIRC they are using vitest so you can prob use assymetric matchers here to future proof those tests and not be annoyed because of a rewording of the message. 

expect.stringMatching or something would do.

Reply:
That should not affect this usecase. There are no side effect code for `formatOffensesJson`. This test would control all input and output from this function so unit test regressions here shouldnt occur from a package bump.